.github/workflows/integration-tests.yml

```
@@ -0,0 +1,78 @@
+name: Integration tests
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  ollama:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+        with:
+          python-version: "3.10.16"
```

I don't think we should be pinning to a Z-Stream of a Python version - would suggest the following instead
```suggestion
          python-version: "3.10"
```